lib/plugin/neovim-plugin.nix

```
@@ -0,0 +1,102 @@
+{
+  lib,
+  nixvimOptions,
+  toLuaObject,
+  nixvimUtils,
+  mkPlugin,
+}:
+with lib;
+rec {
+  mkSettingsOption =
+    {
+      pluginName ? null,
+      options ? { },
+      description ?
+        if pluginName != null then
+          "Options provided to the `require('${pluginName}').setup` function."
+        else
+          throw "mkSettingsOption: Please provide either a `pluginName` or `description`.",
+      example ? null,
+    }:
+    nixvimOptions.mkSettingsOption { inherit options description example; };
+
+  # TODO: DEPRECATED: use the `settings` option instead
+  extraOptionsOptions = {
+    extraOptions = mkOption {
+      default = { };
+      type = with types; attrsOf anything;
+      description = ''
+        These attributes will be added to the table parameter for the setup function.
+        Typically, it can override NixVim's default settings.
+      '';
+    };
+  };
+
+  mkNeovimPlugin =
+    config:
+    {
+      name,
+      defaultPackage, # make this parameter mandatory for neovim plugins
+      # deprecations
+      deprecateExtraOptions ? false,
+      # colorscheme
+      isColorscheme ? false,
+      # options
+      settingsOptions ? { },
+      settingsExample ? null,
```

This could generalize both the settings option description and the callSetup implementation.

```suggestion
      settingsExample ? null,
      settingsDesc ? "Options provided to the `require('${luaName}').${setupFn}` function.",
      setupFn ? "setup",
```

I think it should be possible to override the description too, not just setup name, because some lua plugins may not use a function at all.